Sources/SpringMotionWindowProtocol.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  SpringMotionWindowProtocol.swift
+//
+//
+//  Created by Oryna Semenets on 17.10.2024.
+//
+
+#if canImport(AppKit)
+import AppKit
+
+public protocol SpringMotionWindowProtocol: NSWindow {
+    var configuration: SpringConfiguration { get set }
+    var destinationPoint: NSPoint? { get set }
+    var motionPhysics: SpringMotionPhysics { get set }
+    var currentMotionState: SpringMotionState? { get set }
+    var displayLink: CVDisplayLink? { get set }
+    var anchorWindowFrameObservation: NSKeyValueObservation? { get set }
```

Comment:
I'm sorry, but I don't think that making `displayLink` and the rest of these properties public is a good idea. I'd prefer they be left "under the hood" — clients of these UI components shouldn't need to read or modify them.

I understand that you're trying to minimize code duplication. I think that in our case, the implementation of `SpringMotionPanel` via duplication of `SpringMotionWindow` is the way to go. At least, I can't see any other way that doesn't involve exposing the technical innards of these components.

Reply:
That was my first intention and frirst cmmit 😅 alright, reverted last changes.

Reply:
Nice, thanks :)
Much easier to read now as well, IMO.